--- nss_ldap-207/configure.in.orig	2003-04-25 04:34:05.000000000 +0200
+++ nss_ldap-207/configure.in	2003-05-17 22:24:02.000000000 +0200
@@ -128,7 +128,7 @@
 AC_CHECK_HEADERS(gsssasl.h)
 
 AC_CHECK_LIB(resolv, main)
-AC_CHECK_LIB(nsl, main)
+AC_SEARCH_LIBS(gethostbyname, nsl)
 
 AC_CHECK_FUNCS(strtok_r)
 AC_CHECK_FUNCS(sigset)
@@ -160,8 +160,10 @@
   with_ldap_lib=auto
 fi
 
-AC_CHECK_LIB(dl, dlopen,[LIBS="-ldl $LIBS"],,$LIBS)
-AC_CHECK_LIB(db, main,[LIBS="-ldb $LIBS"],,$LIBS)
+dnl AC_CHECK_LIB(dl, dlopen,[LIBS="-ldl $LIBS"],,$LIBS)
+if test "$enable_rfc2307bis" = "yes" -o "$enable_schema_mapping" = "yes" ; then
+	AC_CHECK_LIB(db, main,[LIBS="-ldb $LIBS"],,$LIBS)
+fi
 dnl Following checks probably not strictly necessary.
 dnl AC_CHECK_LIB(crypto, main,[LIBS="-lcrypto $LIBS"],,$LIBS)
 dnl AC_CHECK_LIB(ssl, main,[LIBS="-lssl $LIBS"],,$LIBS)
