--- nss_ldap-167/Makefile.am.orig	Fri Jun 29 12:17:56 2001
+++ nss_ldap-167/Makefile.am	Mon Aug  6 18:21:45 2001
@@ -48,16 +48,14 @@
 install-exec-local: nss_ldap.so
 	@$(NORMAL_INSTALL)
 if GLIBC
-	-rm -f $(DESTDIR)$(libdir)/$(NSS_LDAP_LIBC_VERSIONED)
-	$(INSTALL_PROGRAM) -o $(INST_UID) -g $(INST_GID) nss_ldap.so $(DESTDIR)$(libdir)/$(NSS_LDAP_LIBC_VERSIONED)
+	$(INSTALL_PROGRAM) nss_ldap.so $(DESTDIR)$(libdir)/$(NSS_LDAP_LIBC_VERSIONED)
 	$(mkinstalldirs) $(DESTDIR)$(libdir)
 	(cd $(DESTDIR)$(libdir); ln -sf $(NSS_LDAP_LIBC_VERSIONED) $(NSS_LDAP_NSS_VERSIONED))
 	$(mkinstalldirs) $(DESTDIR)/usr$(libdir)
 	(cd $(DESTDIR)/usr$(libdir); ln -sf ../..$(libdir)/$(NSS_LDAP_NSS_VERSIONED) .)
 else
-	$(INSTALL_PROGRAM) -o $(INST_UID) -g $(INST_GID) nss_ldap.so $(DESTDIR)$(libdir)/nss_ldap.so.1
-	-rm -f $(DESTDIR)$(libdir)/nss_ldap.so
-	-ln -s $(DESTDIR)$(libdir)/nss_ldap.so.1 $(DESTDIR)$(libdir)/nss_ldap.so
+	$(INSTALL_PROGRAM) nss_ldap.so $(DESTDIR)$(libdir)/nss_ldap.so.1
+	-ln -fs $(DESTDIR)$(libdir)/nss_ldap.so.1 $(DESTDIR)$(libdir)/nss_ldap.so
 endif
 
 endif
@@ -66,9 +64,9 @@
 	@$(NORMAL_INSTALL)
 	@if test ! -f $(DESTDIR)$(sysconfdir)/ldap.conf; then \
 		$(mkinstalldirs) $(DESTDIR)$(sysconfdir); \
-		$(INSTALL_DATA) -o $(INST_UID) -g $(INST_GID) $(srcdir)/ldap.conf $(DESTDIR)$(sysconfdir)/ldap.conf; \
+		$(INSTALL_DATA) $(srcdir)/ldap.conf $(DESTDIR)$(sysconfdir)/ldap.conf; \
 	fi
-	$(INSTALL_DATA) -o $(INST_UID) -g $(INST_GID) $(srcdir)/nsswitch.ldap $(DESTDIR)$(sysconfdir)/nsswitch.ldap; 
+	$(INSTALL_DATA) $(srcdir)/nsswitch.ldap $(DESTDIR)$(sysconfdir)/nsswitch.ldap; 
 
 uninstall-local:
 	@$(NORMAL_UNINSTALL)
